src/cluster.py

```diff
@@ -69,19 +70,18 @@ def add_node_to_cluster(self, event: ops.framework.EventBase) -> None:
 
     def join_node_to_cluster(self, event: ops.framework.EventBase) -> None:
         """Join node to microceph cluster."""
+        logger.debug("handling {event}")
         if not event.unit:
             return
 
-        token = self.charm.peers.get_app_data(f"{event.unit.name}.join_token")
-        if not token:
-            logger.info("Token not available, deferring join event.")
-            event.defer()
-            return
-
         if self.charm.peers.interface.state.joined is True:
             logger.info("Unit has already joined the cluster")
             return
 
+        token = self.charm.peers.get_app_data(f"{event.unit.name}.join_token")
+        if not token:
+            raise sunbeam_guard.BlockedExceptionError(f"join token not found for {gethostname()}")
+
         try:
             microceph.join_cluster(token=token, **self.charm._get_bootstrap_params())
             self.charm.peers.interface.state.joined = True
```

Comment:
Suggest to add debug logging here and at normal exit of method

Reply:
added at line 90